src/core/videosource.h

```diff
@@ -130,7 +130,9 @@ struct FFMS_VideoSource {
     FFMS_Frame *OutputFrame(AVFrame *Frame);
     void SetVideoProperties();
     bool DecodePacket(const AVPacket &Packet);
-    void DecodeNextFrame(int64_t &PTS, int64_t &Pos);
+
+    // Returns the first packet read that has nonzero PTS/DTS (depending on Frames.UseDTS)
+    SmartAVPacket DecodeNextFrame();
```

Comment:
A decode function returning a packet (and not even the one it decoded) seems extremely confusing to read and follow through the code. Preferably it would actually fill the "unique packet index" or hash or whatever in a function argument, which would negate the need for all this smart packet stuff in the first place, and, at least in my opinion, be easier to follow. YMMV though, that is just my personal take/preference.